src/cairo_interaction.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2013-2015, The Rust-GNOME Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+//! Functions to support using cairo
+
+extern crate cairo;
+
+use std::ptr;
+use glib::translate::{ToGlibPtr};
+use libc::{c_double};
+use ffi::{self, GdkRectangle, GdkRGBA};
+use super::{Pixbuf, Window};
+use self::cairo::{Context};
+
+pub fn create(window: &Window) -> Context {
+    unsafe { Context::wrap(ffi::gdk_cairo_create(window.to_glib_none().0)) }
+}
+
+//pub fn create_region_from_surface() { }
+//--> WRAP: gdk_cairo_region_create_from_surface (cairo_surface_t *surface);
+
+//pub fn create_surface_from_pixbuf() { }
+//--> WRAP: gdk_cairo_surface_create_from_pixbuf (const GdkPixbuf *pixbuf, int scale, GdkWindow *for_window);
+
+pub trait GdkCairoInteraction {
+    /// This is a convenience function around `clip_extents()`. It rounds
+    /// the clip extents to integer coordinates and returns a `GdkRectangle`,
+    /// or `None` if no clip area exists.
+    fn get_clip_rectangle(&self) -> Option<GdkRectangle>;
```

Comment:
Sure, the majority of this was written before you changed the way `Rectangle`s are handled, so I figured that we might need a few changes after I merged in those changes.